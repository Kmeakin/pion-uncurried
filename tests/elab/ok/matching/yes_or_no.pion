enum YesOrNo {
    Yes,
    No,
}

let to_bool = fn(x) => match x {
    Yes() => true,
    No() => false,
};

let from_bool = fn(x) => match x {
    true => Yes,
    false => No,
};

let not = fn(x) => match x {
    Yes() => No,
    No() => Yes,
};

let not_yes = not(Yes);
let not_no = not(No);