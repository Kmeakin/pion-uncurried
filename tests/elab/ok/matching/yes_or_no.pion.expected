enum YesOrNo: Type {
    Yes: YesOrNo,
    No: YesOrNo,
}

let to_bool: fn(x: YesOrNo) -> Bool = fn(x: YesOrNo) => match x {
    Yes => true,
    No => false,
};

let from_bool: fn(x: Bool) -> YesOrNo = fn(x: Bool) => match x {
    true => Yes,
    false => No,
};

let not: fn(x: YesOrNo) -> YesOrNo = fn(x: YesOrNo) => match x {
    Yes => No,
    No => Yes,
};

let not_yes: YesOrNo = not(Yes);

let not_no: YesOrNo = not(No);
